superagi/tools/thinking/tools.py

```diff
@@ -0,0 +1,42 @@
+import os
+import openai
+from typing import Type
+
+from pydantic import BaseModel, Field
+from superagi.tools.base_tool import BaseTool
+from superagi.config.config import get_config
+
+# Schema for LLMThinking tool
+
+class LlmTaskSchema(BaseModel):
+    task_description: str = Field(
+        ...,
+        description="Text describing the task for which the GPT model should generate a response.",
+    )
+
+class LlmThinkingTool(BaseTool):
+    name = "LlmThinking"
+    description = (
+        "A tool that interacts with OpenAI's GPT models "
+        "to generate text given a certain task description."
+    )
+    args_schema: Type[LlmTaskSchema] = LlmTaskSchema
+
+    def _execute(self, task_description: str, model_name: str = "text-davinci-002"):
+        api_key = get_config("OPENAI_API_KEY")
+        openai.api_key = api_key
+
+        try:
+            response = openai.Completion.create(
```

Comment:
we are assuming users will use openai. I feel we should abstract this method inside base_llm and open_ai models.